Paused video autoplays when unlock screen after few minutes locked

### Steps to reproduce

1. Play something and leave the video minimized (not sure if the bug is also in other modes) and paused.
2. Lock the screen with the app in foreground and go for a coffee (between 5-10 minutes)
3. Go back and unlock the screen (I use fingerprint)
4. The minimized video starts playing.

Maybe is related to [this](https://github.com/libre-tube/LibreTube/issues/4258#issuecomment-1641554246) but the opposite behavior. 

### Expected behavior

Keep the video paused after unlock for long times

### Actual behavior

It plays when the devices is locked for about >5-10mins

### LibreTube version

0.20.1

### Android version

13

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

When I get back into the app I get crashes like this

```java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.libretube/com.github.libretube.ui.activities.MainActivity}: org.chromium.base.JniException: lateinit property exoPlayer has not been initialized
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3876)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4018)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:111)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2474)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:240)
	at android.os.Looper.loop(Looper.java:351)
	at android.app.ActivityThread.main(ActivityThread.java:8427)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
Caused by: org.chromium.base.JniException: lateinit property exoPlayer has not been initialized
	at com.github.libretube.ui.fragments.PlayerFragment.onDestroy(SourceFile:148)
	at androidx.fragment.app.Fragment.performDestroy(SourceFile:20)
	at androidx.fragment.app.FragmentStateManager.destroy(SourceFile:125)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(SourceFile:370)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(SourceFile:1574)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(SourceFile:82)
	at androidx.fragment.app.FragmentManager.execSingleAction(SourceFile:32)
	at com.github.libretube.helpers.NavigationHelper.navigateVideo$default(SourceFile:171)
	at com.github.libretube.ui.activities.MainActivity.loadIntentData(SourceFile:204)
	at com.github.libretube.ui.activities.MainActivity.onCreate(SourceFile:866)
	at android.app.Activity.performCreate(Activity.java:8422)
	at android.app.Activity.performCreate(Activity.java:8395)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1403)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3849)
	... 12 more

> When I get back into the app I get crashes like this
> 
> ```
> 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3876)
> 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4018)
> 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:111)
> 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
> 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
> 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2474)
> 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 	at android.os.Looper.loopOnce(Looper.java:240)
> 	at android.os.Looper.loop(Looper.java:351)
> 	at android.app.ActivityThread.main(ActivityThread.java:8427)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
> Caused by: org.chromium.base.JniException: lateinit property exoPlayer has not been initialized
> 	at com.github.libretube.ui.fragments.PlayerFragment.onDestroy(SourceFile:148)
> 	at androidx.fragment.app.Fragment.performDestroy(SourceFile:20)
> 	at androidx.fragment.app.FragmentStateManager.destroy(SourceFile:125)
> 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(SourceFile:370)
> 	at androidx.fragment.app.FragmentManager.executeOpsTogether(SourceFile:1574)
> 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(SourceFile:82)
> 	at androidx.fragment.app.FragmentManager.execSingleAction(SourceFile:32)
> 	at com.github.libretube.helpers.NavigationHelper.navigateVideo$default(SourceFile:171)
> 	at com.github.libretube.ui.activities.MainActivity.loadIntentData(SourceFile:204)
> 	at com.github.libretube.ui.activities.MainActivity.onCreate(SourceFile:866)
> 	at android.app.Activity.performCreate(Activity.java:8422)
> 	at android.app.Activity.performCreate(Activity.java:8395)
> 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1403)
> 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3849)
> 	... 12 more
>     
> ```

I got errors sometimes but cant reproduce it always, so didnt posted any log here

--- a/app/src/main/java/com/github/libretube/constants/IntentData.kt
+++ b/app/src/main/java/com/github/libretube/constants/IntentData.kt
@@ -41,4 +41,5 @@ object IntentData {
     const val maxAudioQuality = "maxAudioQuality"
     const val audioLanguage = "audioLanguage"
     const val captionLanguage = "captionLanguage"
+    const val wasIntentStopped = "wasIntentStopped"
 }

--- a/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
+++ b/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
@@ -113,7 +113,6 @@ import com.github.libretube.util.PlayingQueue
 import com.github.libretube.util.TextUtils
 import com.github.libretube.util.TextUtils.toTimeInSeconds
 import com.github.libretube.util.YoutubeHlsPlaylistParser
-import java.util.*
 import java.util.concurrent.Executors
 import kotlin.math.abs
 import kotlinx.coroutines.Dispatchers
@@ -782,6 +781,11 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
         ) {
             exoPlayer.pause()
         }
+
+        // the app was put somewhere in the background - remember to not automatically continue
+        // playing on re-creation of the app
+        requireArguments().putBoolean(IntentData.wasIntentStopped, true)
+
         super.onPause()
     }
 
@@ -948,7 +952,11 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
 
             initializePlayerView()
 
-            exoPlayer.playWhenReady = PlayerHelper.playAutomatically
+            // don't continue playback when the fragment is re-created after Android killed it
+            val wasIntentStopped = requireArguments().getBoolean(IntentData.wasIntentStopped, false)
+            exoPlayer.playWhenReady = PlayerHelper.playAutomatically && !wasIntentStopped
+            requireArguments().putBoolean(IntentData.wasIntentStopped, false)
+
             exoPlayer.prepare()
 
             if (binding.playerMotionLayout.progress != 1.0f) {

